### PR TITLE
Clean metadata && install bind-licence separately (?)

### PR DESCRIPTION
**Pull request by kvch**
CC @simitt 

Fix the ARM build issues because `bind-license`

### CODE REVIEW

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 1]
```diff
@@ -30,8 +30,24 @@ FROM {{ .from }}
 {{- if contains .from "ubi-minimal" }}
 RUN for iter in {1..10}; do microdnf update -y && microdnf install -y shadow-utils jq &&  microdnf clean all && exit_code=0 && break || exit_code=$? && echo "microdnf error: retry $iter in 10s" && sleep 10; done; (exit $exit_code)
 {{- else }}
+
+# Installing jq needs to be installed after epel-release and cannot be in the same yum install command.
```
Comment (kvch) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:34]: I will remove this unnecessary comment once we validated the fix works.

**Review by ph — APPROVED** [review]
@kvch changes looks good to me, I've ran `make snapshot` and everything seems to be build correctly, anything I am missing to test the behavior?

**Review by simitt — APPROVED** [review]
generally LGTM

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 2]
```diff
@@ -30,8 +30,27 @@ FROM {{ .from }}
...
+  esac ;
+
 # Installing jq needs to be installed after epel-release and cannot be in the same yum install command.
-RUN for iter in {1..10}; do yum update --setopt=tsflags=nodocs -y && yum install --setopt=tsflags=nodocs -y epel-release &&  yum clean all && exit_code=0 && break || exit_code=$? && echo "yum error: retry $iter in 10s" && sleep 10; done; (exit $exit_code)
+RUN for iter in {1..10}; do yum update --setopt=tsflags=nodocs -y \\
+    && yum install --setopt=tsflags=nodocs -y epel-release \\
+    && yum clean all \\
+    && yum clean metadata \\
+    && exit_code=0 && break || exit_code=$? && echo "yum error: retry $iter in 10s" && sleep 10; \\
+done; (exit $exit_code)
```
Comment (jsoriano) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:53]: Previous CI build failed building the elastic-agent docker image. Are these double back-slashes needed?
```suggestion
RUN for iter in {1..10}; do yum update --setopt=tsflags=nodocs -y \
    && yum install --setopt=tsflags=nodocs -y epel-release \
    && yum clean all \
    && yum clean metadata \
    && exit_code=0 && break || exit_code=$? && echo "yum error: retry $iter in 10s" && sleep 10; \
done; (exit $exit_code)
```

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 3]
```diff
@@ -30,8 +30,27 @@ FROM {{ .from }}
 {{- if contains .from "ubi-minimal" }}
 RUN for iter in {1..10}; do microdnf update -y && microdnf install -y shadow-utils jq &&  microdnf clean all && exit_code=0 && break || exit_code=$? && echo "microdnf error: retry $iter in 10s" && sleep 10; done; (exit $exit_code)
 {{- else }}
+
+RUN set -e ; \
+  case "$(arch)" in \
+    aarch64) \
+        for iter in {1..10}; do yum update --setopt=tsflags=nodocs -y \
+            && yum install --setopt=tsflags=nodocs -y http://mirror.centos.org/centos/7/updates/aarch64/Packages/bind-license-9.11.4-26.P2.el7_9.5.noarch.rpm \
```
Comment (jsoriano) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:38]: When this is merged please create a follow-up issue to clean this.

**Review by jsoriano — APPROVED** [review]
Thanks!

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl (hunk 2 above)
Comment (jsoriano) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:53]: This was the failure I am refering to:
```
[2021-05-11T19:38:42.313Z] Error: failed building elastic-agent type=docker for platform=linux/amd64: failed to build docker: running "docker build -t docker.elastic.co/beats/elastic-agent:8.0.0-SNAPSHOT build/package/elastic-agent/elastic-agent-linux-amd64.docker/docker-build" failed with exit code 1
```

Comment (kvch) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:53]: No, they do not. However, we never get to this point, so I did not bother to clean this up yet.

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl (hunk 3 above)
Comment (kvch) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:38]: Sure, I hope to remove this terrible hack as soon as possible.

**Comment by kvch** [comment]
I moved the PR to draft as the fix that works for everyone, does not help us. I have to track down what causes the problems for us.

**Comment by kvch** [comment]
Closing in favour of #25679

**Pull request closed without merging**